src/Symfony/Component/Notifier/Bridge/Telegram/TelegramTransport.php

```diff
@@ -52,17 +52,21 @@ public function __construct(
         private ?string $chatChannel = null,
         ?HttpClientInterface $client = null,
         ?EventDispatcherInterface $dispatcher = null,
+        private readonly bool $disableHttps = false,
     ) {
         parent::__construct($client, $dispatcher);
     }
 
     public function __toString(): string
     {
-        if (null === $this->chatChannel) {
-            return \sprintf('telegram://%s', $this->getEndpoint());
+        $toString = \sprintf('telegram://%s', $this->getEndpoint());
+        $formattedOptions = http_build_query(['channel' => $this->chatChannel, 'disable_https' => $this->disableHttps ?: null]);
+
+        if ($formattedOptions) {
```

Comment:
something is wrong here: the condition is always true now

Reply:
It will be true only if one of the `$chatChannel` or `$disableHttps` properties has been set to something except the default. Unless this, the condition will always be false.

`echo http_build_query(['channel' => null, 'disable_https' => false ?: null]);` should return an empty string which will not pass the condition.

I may use `empty()` function in the condition for better readability? (e.g. `if (!empty($formattedOptions))`)

Reply:
Tests are passing well also

Reply:
I just tested it, as I was always confused, but http_build_query behaves like `array_filter` and if chatChannel and disableHttps is null, `$formattedOptions` is not true

Reply:
Actually, I don't get what is wrong with the `http_build_query` or the `if` statement 😅. It just tries to build a query string from a given set of class properties ($chatChannel & $disableHttps). I also have seen many other bridges use `http_build_query` in the same function to build the transport DSN.

If that's not a good approach, what is the alternative then?

Reply:
[1](https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/Notifier/Bridge/ClickSend/ClickSendTransport.php#L46C1-L55C6) [2](https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/Notifier/Bridge/Slack/SlackTransport.php#L50)